discovery-provider/src/tasks/index_rewards_manager.py

```diff
@@ -315,9 +365,27 @@ def process_batch_sol_reward_manager_txs(
                 )
             )
 
+            prebalance = transfer_instr["prebalance"]
+            postbalance = transfer_instr["postbalance"]
+            balance_change = postbalance - prebalance
+            audio_tx_histories.append(
+                AudioTransactionsHistory(
+                    user_bank=user_bank_map[eth_recipient],
+                    slot=tx["slot"],
+                    signature=tx["tx_sig"],
+                    transaction_type=challenge_type_map[challenge_id],
+                    method=TransactionMethod.receive,
+                    transaction_created_at=timestamp,
+                    change=Decimal(balance_change),
+                    balance=Decimal(postbalance),
+                    tx_metadata=challenge_id,
+                )
+            )
+
         if challenge_disbursements:
-            # Save out the disbursements
+            # Save out the disbursements and transaction history list
             session.bulk_save_objects(challenge_disbursements)
+            session.bulk_save_objects(audio_tx_histories)
```

Comment:
curious if appending the arrays together and calling bulk_save_objects a single time would be quicker (I think probably not since it flushes everything when the session goes out of scope?)

Reply:
we could take advantage of python magic and just concat them like `session.bulk_save_objects(challenge_disbursements + audio_tx_histories)` 👀

Reply:
genius

Reply:
Getting this: `Unsupported operand types for + ("List[ChallengeDisbursement]" and "List[AudioTransactionsHistory]"`